packages/cli/src/config/settingsSchema.ts

```diff
@@ -1203,6 +1203,37 @@ const SETTINGS_SCHEMA = {
           },
         },
       },
+      allowedEnvironmentVariables: {
+        type: 'array',
+        label: 'Allowed Environment Variables',
+        category: 'Security',
+        requiresRestart: true,
+        default: [] as string[],
+        description:
+          'Environment variables to always allow (bypass redaction).',
+        showInDialog: false,
+        items: { type: 'string' },
+      },
+      blockedEnvironmentVariables: {
+        type: 'array',
+        label: 'Blocked Environment Variables',
+        category: 'Security',
+        requiresRestart: true,
+        default: [] as string[],
+        description: 'Environment variables to always redact.',
+        showInDialog: false,
+        items: { type: 'string' },
+      },
+      enableEnvironmentVariableRedaction: {
+        type: 'boolean',
+        label: 'Enable Environment Variable Redaction',
+        category: 'Security',
+        requiresRestart: true,
+        default: true,
```

Comment:
I think we should default this to false, until we figure out how to onboard users seamlessly (not a requirement for this PR)